blob/azureblob/azureblob.go

```diff
@@ -120,54 +90,123 @@ const (
 	defaultUploadBlockSize          = 8 * 1024 * 1024 // configure the upload buffer size
 )
 
-func init() {
-	blob.Register("azblob", openURL)
+// Scheme is the URL scheme conventionally used for Azure Storage in a URLMux.
+const Scheme = "azblob"
+
+// URLOpener opens Azure URLs like "azblob://mybucket".
```

Comment:
I'm concerned that this is making what I think will be a common case of just wanting to use a URL (with creds pulled from the environment) too difficult.

It looks like this:

```
m := map[string]blob.BucketURLOpener{
  gcsblob.Scheme : gcsblob.URLOpener{AllowURLOverrides: true},
}
mux := blob.NewURLMux(m)
...
```

The user has to figure out to use `gcsblob.Scheme`, and `gcsblob.URLOpener`, and notice the `AllowURLOverrides` field but also know that they can ignore the other fields.

Alternative: can we have the per-provider URLOpener` type be private, and have `URLOptions` be the public type (if needed for the provider)?

The exposed API would be something like:

```
m := map[string]blob.BucketURLOpener{}
gcsblob.RegisterURL(m, true /* allow environment-dependent stuff */, &gcsblob.URLOptions{} /* or nil */)
mux := blob.NewURLMux(m)
...
```

`RegisterURL` would create the `URLOpener` struct.

Or maybe the boolean argument can be dropped entirely, and be made implicit based on whether a non-nil `Options` is used? I.e., you either provide a `URLOptions` struct at `Register` time, in which case it will be used, or you provide `nil` and one will be created from the URL parameters. Or do you think it's useful to allow a combination (e.g., some things set from the user-provided struct, others filled in via the URL)? That might be confusing.